debian/rules

```diff
@@ -1,114 +1,25 @@
 #!/usr/bin/make -f
-# -*- makefile -*-
-# Sample debian/rules that uses debhelper.
-# GNU copyright 1997 to 1999 by Joey Hess.
 
-# Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
-CFLAGS = -Wall -Wno-format -g
+export DEB_CFLAGS_MAINT_APPEND  = -Wall -Wno-format -Wno-format-security -g -DTAP_AND_HOLD
+export DEB_LDFLAGS_MAINT_APPEND = -Wl,--as-needed
 
-ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
-	CFLAGS += -O0
-else
-	CFLAGS += -O2
-endif
-ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
-	INSTALL_PROGRAM += -s
-endif
-ifneq (,$(findstring thumb,$(DEB_BUILD_OPTIONS)))
-        CFLAGS += -mthumb
-endif
 ifeq (,$(findstring nolauncher,$(DEB_BUILD_OPTIONS)))
 	CONFIGURE_OPTIONS += --enable-maemo-launcher
 	USE_MAEMO_LAUNCHER = true
 endif
 
-CFLAGS += -DTAP_AND_HOLD
-
-CXXFLAGS = $(CFLAGS)
-
-config.status:
-	if [ ! -x configure ]; then ./autogen.sh; fi
-	dh_testdir
-	# Add here commands to configure the package.
-	CFLAGS="$(CFLAGS)" CXXFLAGS="$(CXXFLAGS)" ./configure --prefix=/usr --sysconfdir=/etc --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info --disable-static $(CONFIGURE_OPTIONS)
-
-
-build: build-stamp
-
-build-stamp:  config.status
-	dh_testdir
-
-	# Add here commands to compile the package.
-	$(MAKE)
+%:
+	dh $@
 
-	touch build-stamp
+override_dh_configure:
+	dh_configure -- $(CONFIGURE_OPTIONS)
 
-clean:
-	dh_testdir
-	dh_testroot
-	rm -f build-stamp 
-
-	# Add here commands to clean up after the build process.
-	-$(MAKE) distclean
-
-	dh_clean 
-
-install: build
-	dh_testdir
-	dh_testroot
-	dh_clean -k 
-	dh_installdirs
-
-        # Add here commands to install the package into debian/hildon-application-manager.
-	$(MAKE) install DESTDIR=$(CURDIR)/debian/hildon-application-manager
+override_dh_install:
+	dh_install
 	for i in alpha beta; do \
 	  echo $$i; \
 	  mkdir -p $(CURDIR)/debian/hildon-application-manager/usr/share/themes/$$i/backgrounds; \
 	  cp share/themes/$$i-background.png $(CURDIR)/debian/hildon-application-manager/usr/share/themes/$$i/backgrounds/applicationmanager-portrait.png; \
 	done
-	rm `find $(CURDIR)/debian/hildon-application-manager -name '*.la'`
-
-
-# Build architecture-independent files here.
-binary-indep: build install
-# We have nothing to do by default.
-
-# Build architecture-dependent files here.
-binary-arch: build install
-	dh_testdir
-	dh_testroot
-#	dh_installchangelogs ChangeLog
-#	dh_installdocs
-#	dh_installexamples
-#	dh_install
-#	dh_installmenu
-#	dh_installdebconf	
-#	dh_installlogrotate
-#	dh_installemacsen
-#	dh_installpam
-#	dh_installmime
-#	dh_installinit -r -- start 52 2 3 4 . stop 20 0 1 5 6 .
-#	dh_installcron
-#	dh_installinfo
-#	dh_installman
-	dh_installxsession -u 'post 40'
-	dh_link
-	if test "x$(USE_MAEMO_LAUNCHER)" = "xtrue"; then \
-	  dh_maemolauncher -v; \
```

Comment:
Why dh_maemolauncher is not used anymore?

Reply:
AFAIK, it's used for injecting dependencies. We don't need this as they're properly handled in debian/control.

Reply:
What dependencies? Could you explain a bit more what exactly it does?

Reply:
https://github.com/maemo-leste/maemo-launcher/blob/master/debian/dh_maemolauncher

Reply:
Hm... What is this code for?
```
doit("mv", "-f", "$tmp/$so", "$tmp/$so.launch");
doit("ln", "-s", "/usr/bin/maemo-invoker", "$tmp/$so");
```

Reply:
Reverted in 780bdf22fd425ba5acfd175baf70609250d7907c though. We can use it as then we'll have the .launch binary.

Reply:
That code creates a .launch binary and instead launches binaries with maemo-invoker. See http://maemo.org/maemo_release_documentation/maemo4.1.x/node6.html for what maemo-invoker does.

The same thing is happening in hildon-desktop, and IIRC @MerlijnWajer had some problems with starting h-d without maemo-invoker.

Reply:
Seems like we need it.

Reply:
ACK. I reintroduced it and it'll be like it never went away when squashed :)